src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -45,32 +51,53 @@ public User updateUser(@RequestBody User user, @PathVariable Long id) {
         return userRepository.save(user);
     }
 
-    @PostMapping("/addBooks")
-    public void addBooks(Book book, Long idUser){
-        User userAux= findOne(idUser);
-        for (Book bookAux: userAux.getBooks()) {
 
-            if(bookAux.getIsbn() == book.getIsbn()){
-                throw new RuntimeException();
+    @PutMapping("/addBooks/{idUser}/{idBook}")
+    public User addBooks(@RequestBody User user, @PathVariable("idUser") Long idUser, @PathVariable("idBook") Long idBook) throws BookAlreadyOwnedException{
+
+        if (user.getId() != idUser) {
+            throw new RuntimeException();
+        }
+
+        User userAux= userRepository.findById(idUser).orElseThrow(RuntimeException::new);
+
+        Book bookAux= bookRepository.findById(idBook).orElseThrow(RuntimeException::new);
+
+
+        for (Book b: userAux.getBooks()) {
+
+            if(bookAux.getIsbn() == b.getIsbn()){
+                throw new BookAlreadyOwnedException("Error: The book you are trying to add already exists.");
             }
         }
 
-        userAux.addBook(book);
+        userAux.addBook(bookAux);
+        return updateUser(userAux,idUser);
```

Comment:
remember not to use other controllers methods internally. 
You should do `userRepository.save(userAux)`